src/app/ombul/[...yearAndName]/page.tsx

```diff
@@ -0,0 +1,85 @@
+import styles from './page.module.scss'
+import ChangeName from './ChangeName'
+import OmbulAdmin from './OmbulAdmin'
+import { readOmbul } from '@/actions/ombul/read'
+import PdfDocument from '@/components/PdfDocument/PdfDocument'
+import { requireUser } from '@/auth'
+import SlideInOnView from '@/app/components/SlideInOnView/SlideInOnView'
+import EditableTextField from '@/app/components/EditableTextField/EditableTextField'
+import { updateOmbul } from '@/actions/ombul/update'
+import Link from 'next/link'
+import { notFound } from 'next/navigation'
+
+type PropTypes = {
+    params: {
+        yearAndName: string[]
+    }
+}
+
+export default async function Ombul({ params }: PropTypes) {
+    const year = parseInt(decodeURIComponent(params.yearAndName[0]), 10)
+    const name = decodeURIComponent(params.yearAndName[1])
+    if (!year || !name || params.yearAndName.length > 2) notFound()
+    const ombulRes = await readOmbul({
+        name,
+        year
+    })
+    if (!ombulRes.success) notFound()
+    const ombul = ombulRes.data
+
+    const path = `/store/ombul/${ombul.fsLocation}`
+
+    const user = await requireUser({
```

Comment:
This should be called before readOmbul. ReadOmbul will throw an error and then notFound will be called, before we can check this.